lib/internal/repl.js

```diff
@@ -51,19 +53,45 @@ function createRepl(env, cb) {
   }
 
   const repl = REPL.start(opts);
-  if (opts.terminal && env.NODE_REPL_HISTORY_FILE) {
-    return setupHistory(repl, env.NODE_REPL_HISTORY_FILE, cb);
+  if (opts.terminal && env.NODE_REPL_HISTORY !== '') {
+    return setupHistory(repl, env.NODE_REPL_HISTORY,
+                        env.NODE_REPL_HISTORY_FILE, cb);
   }
   repl._historyPrev = _replHistoryMessage;
   cb(null, repl);
 }
 
-function setupHistory(repl, historyPath, ready) {
-  const fs = require('fs');
+function setupHistory(repl, historyPath, oldHistoryPath, ready) {
+  if (!historyPath) {
+    try {
+      historyPath = path.join(os.homedir(), '.node_repl_history');
+    } catch (err) {
+      console.error('Error: REPL could not get the home directory.');
+      ready(null, repl);
```

Comment:
I think this should be fine — I might expand the error to say "REPL history will not be persisted for this session."